File: 프로그래머스/Level1/시저 암호.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s, int n) {
    string answer = "";
    
    for(int i=0; i<s.length(); i++){
        if(s[i] != ' '){
            char st = ('A'<= s[i] && s[i] <='Z') ? 'A' : 'a';
            answer += st + (s[i] + n - st)%26;
        }
        else if(s[i] == ' '){
            answer += ' ';
        }
    }
    return answer;
}

File: 프로그래머스/Level1/시저 암호.js
function solution(s, n) {
    let answer = s.split('');
    let result = "";
    
    for(let i=0; i<answer.length; i++){
        if(answer[i] === ' '){
            result += " ";
            continue;
        }
        else{
            const ascii = parseInt(answer[i].charCodeAt(0), 10);
            const tmp = (ascii >= 65 && ascii <= 90) ? 65 : 97;
            
            result += String.fromCharCode(tmp + (ascii-tmp+n)%26);
        }
    }
    return result;
}
